ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/QueryStreamWriter.java

```diff
@@ -94,24 +96,24 @@ public void write(OutputStream out) throws IOException {
         }
         drainAndThrowOnError(out);
       }
-    } catch (EOFException exception) {
+
+      drain(out);
+
+      if (limitReached) {
+        objectMapper.writeValue(out, StreamedRow.finalMessage("Limit Reached"));
+        out.write("\n".getBytes(StandardCharsets.UTF_8));
+        out.flush();
+      }
+    } catch (final EOFException exception) {
       // The user has terminated the connection; we can stop writing
       log.warn("Query terminated due to exception:" + exception.toString());
-    } catch (InterruptedException exception) {
+    } catch (final InterruptedException exception) {
       // The most likely cause of this is the server shutting down. Should just try to close
       // gracefully, without writing any more to the connection stream.
       log.warn("Interrupted while writing to connection stream");
```

Comment:
Should we also try to send this message to the user? Something like "Server shut down, stopping query".

Reply:
Maybe - but in a different PR...